core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -624,7 +624,8 @@ private[spark] object Utils extends Logging {
       case _ =>
         val fs = getHadoopFileSystem(uri, hadoopConf)
         val path = new Path(uri)
-        fetchHcfsFile(path, new File(targetDir, path.getName), fs, conf, hadoopConf, fileOverwrite)
+        fetchHcfsFile(path, targetDir, fs, conf, hadoopConf, fileOverwrite,
+                      Some(filename))
```

Comment:
Oh actually this doesn't follow the convention for named params (`filename = Some(filename)`). I'll fix it, and in that case it doesn't fit in the previous line...